node/src/chain_spec/mod.rs

```diff
@@ -208,57 +168,69 @@ pub fn testnet_genesis(
         ImOnlineId,
         AuthorityDiscoveryId,
     )>,
+    initial_nominators: Vec<AccountId>,
     root_key: AccountId,
-    endowed_accounts: Vec<AccountId>,
+    mut endowed_accounts: Vec<AccountId>,
     members: Vec<membership::genesis::Member<u64, AccountId>>,
-    forum_config: ForumConfig,
-    initial_balances: Vec<(AccountId, Balance)>,
-    content_config: ContentConfig,
+    forum_cfg: ForumConfig,
+    genesis_balances: Vec<(AccountId, Balance)>,
+    content_cfg: ContentConfig,
 ) -> GenesisConfig {
-    const STASH: Balance = 5_000;
-    const ENDOWMENT: Balance = 100_000_000;
+    // endow all authorities and nominators.
+    initial_authorities
+        .iter()
+        .map(|x| &x.0)
+        .chain(initial_nominators.iter())
+        .for_each(|x| {
+            if !endowed_accounts.contains(x) {
+                endowed_accounts.push(x.clone())
+            }
+        });
+
+    // stakers: all validators and nominators.
+    let mut rng = rand::thread_rng();
+    let stakers = initial_authorities
+        .iter()
+        .map(|x| (x.0.clone(), x.1.clone(), STASH, StakerStatus::Validator))
+        .chain(initial_nominators.iter().map(|x| {
+            use rand::{seq::SliceRandom, Rng};
+            let limit = (MaxNominations::get() as usize).min(initial_authorities.len());
+            let count = rng.gen::<usize>() % limit;
+            let nominations = initial_authorities
```

Comment:
This can generate empty set of nominations and cause the runtime to fail with:
```
Thread 'main' panicked at 'Expected Ok(_). Got Err(
    Module(
        ModuleError {
            index: 8,
            error: [
                4,
                0,
                0,
                0,
            ],
            message: Some(
                "EmptyTargets",
            ),
        },
    ),
)'
```

(tested with both 1 and 2 authorities)